pysquared/hardware/radio/manager/base.py

```diff
@@ -110,6 +110,11 @@ def set_modulation(self, modulation: Type[RadioModulation]) -> None:
                 current=current_modulation,
             )
 
+    def modify_config(self, radio_config: RadioConfig) -> None:
+        """Modify the radio configuration. This will apply the new configuration and reinitialize the radio."""
+        self._radio_config = radio_config
+        self._initialize_radio(self.get_modulation())
```

Comment:
We may be able to loosen these rules now that we have more memory but, as of right now, we're not allowing the radio to be reinitialized during runtime. Doing so makes understanding the memory state of the satellite more challenging.

Instead my recommendation is to implement these in the individual radio managers and allow them to modify any variables that we are allowed to set outside of initialization.